chef/cookbooks/ceph/recipes/osd.rb

```diff
@@ -127,7 +137,7 @@
         end
         device["device"] = d.name
         node.set["ceph"]["osd_devices"].push(device)
-        node.save
+        dirty = true
```

Comment:
Yes, the array of devices is enumerable (i.e. usable) immediately below even after moving node.save to the end of the file. The node.save is only needed to persist the values for subsequent executions of the cookbook.